pkg/osdCloud/common.go

```diff
@@ -0,0 +1,215 @@
+package osdCloud
```

Comment:
files with names like `common.go` are usually an antipattern.

Could we abstract this further into `pkg/osdCloud/interfaces.go` `pkg/osdCloud/gcp.go` `pkg/osdCloud/aws.go`?

Also, instead of `osdCloud` maybe something like `cloudprovider`?  I think the camelCase will come back to bite us with package naming in the future.

Reply:
There are other camelCase package names in osdctl already, so I'd leave it as it was for now.

But I moved the implementations of the clients to the respective files and only left the base types in interfaces.go